nl_server/config_reader.py

```diff
@@ -94,13 +96,26 @@ def merge_vertex_ai_configs(m: ModelConfig,
 
 
 def read_catalog(catalog_paths: List[str] = _DEFAULT_CATALOG_PATHS,
-                 catalog_dict: Dict = None) -> Catalog:
+                 catalog_dict: Dict = None,
+                 additional_catalog_path: str = None) -> Catalog:
   """Reads the catalog from the config files and merges them together.
   """
   partial_catalogs = []
+  if additional_catalog_path:
+    catalog_paths.append(additional_catalog_path)
   for p in catalog_paths:
     all_paths = []
-    if os.path.exists(p):
+
+    # If gcs path, download it locally.
+    if gcs.is_gcs_path(p):
+      gcs_path = p
+      logging.info('Loading catalog from gcs path: %s', gcs_path)
+      p = gcs.maybe_download(gcs_path)
+      if not p:
+        logging.info('GCS catalog path not found and will be skipped: %s',
+                     gcs_path)
```

Comment:
I wonder if should we have this gcs.maybe_download in "admin.py"?  So it downloads locally and NL server or build_embeddings need not?

Reply:
It would need to be downloaded by nl server at startup.